.gitlab-ci.yml

```
@@ -187,6 +187,54 @@ spellcheck:
         cargo spellcheck check -vvv --cfg=.config/cargo_spellcheck.toml --checkers hunspell --code 1 examples/delegator/${contract}/;
       done
 
+codecov:
+  stage:                           workspace
+  <<:                              *docker-env
+  <<:                              *test-refs
+  needs:
+    - job:                         check-std
+      artifacts:                   false
+  variables:
+    # For codecov it's sufficient to run the fuzz tests only once.
+    QUICKCHECK_TESTS:              1
+    # Now we're using the MIR source-based Rust test coverage, the setup assumes that default
+    # toolchain is nightly and both `grcov` and `rustup component add llvm-tools-preview` are
+    # installed on it.
+    RUSTFLAGS:                     "-Zinstrument-coverage"
+    LLVM_PROFILE_FILE:             "llvmcoveragedata-%p-%m.profraw"
+    CARGO_INCREMENTAL:             0
+    # The `cargo-taurpalin` coverage reporting tool seems to have better code instrumentation and thus
+    # produces better results for Rust codebases in general. However, unlike `grcov` it requires
+    # running docker with `--security-opt seccomp=unconfined` which is why we use `grcov` instead.
+  before_script:
+    - *rust-info-script
+    # RUSTFLAGS are the cause target cache can't be used here
```

```suggestion
    # RUSTFLAGS are the reason `target` cache can't be used here
```
Is this still relevant since we've changed all the `RUSTFLAGS` being used

---

Don't know actually, gotta try.